lib/sudo-pam/src/lib.rs

```diff
@@ -472,21 +459,17 @@ impl<'a, C: Converser> Drop for PamContext<'a, C> {
     fn drop(&mut self) {
         // data_ptr's pointee is de-allocated in this scope
         let _data = unsafe { Box::from_raw(self.data_ptr) };
-        if self.pamh.is_some() {
-            if !self.session_started {
-                let _ = self.close_session();
-            }
-
-            // It looks like PAM_DATA_SILENT is important to set for our sudo context, but
-            // it is unclear what it really does and does not do, other than the vague
-            // documentation description to 'not take the call to seriously'
-            // Also see https://github.com/systemd/systemd/issues/22318
-            unsafe {
-                pam_end(
-                    self.handle().unwrap(),
-                    self.last_pam_status.unwrap_or(PAM_SUCCESS as libc::c_int),
-                ) | PAM_DATA_SILENT as i32
-            };
-        }
+        let _ = self.close_session();
+
+        // It looks like PAM_DATA_SILENT is important to set for our sudo context, but
+        // it is unclear what it really does and does not do, other than the vague
+        // documentation description to 'not take the call to seriously'
+        // Also see https://github.com/systemd/systemd/issues/22318
+        unsafe {
+            pam_end(
+                self.pamh,
+                self.last_pam_status.unwrap_or(PAM_SUCCESS as libc::c_int) | PAM_DATA_SILENT as i32,
```

Comment:
I would make a temp var for flags to make the code clearer but anyway thanks for fixing it.